docs/user/migration/external-name-alias.md

```diff
@@ -0,0 +1,154 @@
+# Migration guide for external name alias behavior change
+
+## Context
+
+Istio with 1.21 version has changed the behavior of the Service of type ExternalName.
+Previously, for the Service of type ExternalName a separate `cluster` in the Envoy config was created.
+With the change, the Service of type ExternalName is treated as an alias of the Service that it points to.
+This change has been introduced to align the behavior of Istio with handling of the Service of type ExternalName with the Kubernetes behavior.
+Since it caused some issues for our users, we introduced a new annotation "disable-external-name-alias" to disable this behavior.
+Under the hood, we apply `ENABLE_EXTERNAL_NAME_ALIAS` environment variable on the Istiod component.
+At certain point Istio will drop the support for this environment variable, making the behavior change obligatory.
+Currently, it's already removed from the main branch with the [PR](https://github.com/istio/istio/pull/52317).
+We expect it to be a part of the 1.24 release.
+Due to that, deprecation on the annotation will be introduced.
+If you are affected by this change, you should update your configuration to align with the new behavior.
+
+## Virtual Service
+
+Due to the current Istio misbehavior, VirtualService cannot point with its destination to the Service of type ExternalName as the upgrade notes states [Ref](https://istio.io/latest/news/releases/1.21.x/announcing-1.21/upgrade-notes/).
+The limitation exists in the traffic coming from the internal mesh clients.
+Creating a ServiceEntry, has no effect despite upgrade notes for 1.21.
+Routing to the Service of type ExternalName which alias other type of Service also will not work.
+Instead, it has to point to the actual host that is being aliased by the Service of type ExternalName.
+With [PR](https://github.com/istio/istio/pull/52589) merged, VirtualService should be able to point to the Service of type ExternalName for the internal mesh traffic.
```

Comment:
I would not mention that in documentation, the state of the PR might change, and anyway it needs to have a release and we need to bump that

Reply:
Yeah, that's right. I'm only concerned about the context. Certain parts of the migration guide might not be fully reflecting the state after the fix. Do you have any suggestions how'd you handle it here?

Reply:
Should we somehow mention, that the current steps are adjusted to the Istio not intended behavior atm?

Reply:
let's proceed as was discussed